Add gitlabci.yml file

cmft opened the pull request:
Add gitlabci.yml file for CI/packaging purposes.

cmft committed:
Add .gitlab-ci.yml file

cmft committed:
Remove print

cpascual commented:
The same comments I did in https://github.com/ALBA-Synchrotron/VacuumController/pull/3 are pertinent here

cmft committed:
Hide some specific ALBA jobs

upload_to_staging and promote_to_production job are ALBA specific avoid their execution out of ALBA

cpascual commented on .gitlab-ci.yml:
```diff
@@ -0,0 +1,147 @@
+  DOCKER_DRIVER: overlay2
+  ARTIFACTS_DIR: "$CI_PROJECT_DIR/dist"
+  # Set up variables for pushing the generated tags
+  DEPLOY_GIT_REPO: "git.cells.es/$CI_PROJECT_PATH.git"
```
do not hardcode alba specific stuff in DEPLOY_GIT_REPO

cpascual commented on .gitlab-ci.yml:
```diff
@@ -0,0 +1,147 @@
+  # ALBA HACK:
+  GIT_SSL_NO_VERIFY: 'true'
+  DEBFULLNAME: "CI Boot"
+  DEBEMAIL: "controls@cells.es"
```
do not hardcode alba specific stuff in DEBEMAIL

cmft committed:
Avoid to hardcode ALBA variables
